test/e2e/container_iface_name_test.go

```diff
@@ -0,0 +1,600 @@
+	containersConf := []byte("[containers]\ninterface_name = \"device\"\n")
+	err := os.WriteFile(configPath, containersConf, os.ModePerm)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Set custom containers.conf file
+	os.Setenv("CONTAINERS_CONF", configPath)
+	if IsRemote() {
+		pTest.RestartRemoteService()
+	}
+}
+
+var _ = Describe("Podman container interface name", func() {
+
+	It("podman container interface name with default scheme for bridge network", func() {
+		// Assert that the network interface name inside container for
+		// bridge network is ethX when interface_name in the
+		// containers.conf file is set to default value, i.e., "".
+
+		netName1 := createNetworkName("bridge")
+		netName2 := createNetworkName("bridge")
+
+		defer podmanTest.removeNetwork(netName1)
+		nc1 := podmanTest.Podman([]string{"network", "create", netName1})
+		nc1.WaitWithDefaultTimeout()
+		Expect(nc1).Should(ExitCleanly())
+
+		// Inspect the network configuration
+		nic1 := podmanTest.Podman([]string{"network", "inspect", netName1, "--format", "{{.NetworkInterface}}"})
+		nic1.WaitWithDefaultTimeout()
+		Expect(nic1).Should(ExitCleanly())
+
+		// Once a container executes a new network, the nic will be
+		// created. We should clean those up best we can
+		defer deleteNetworkDevice(nic1.OutputToString())
+
+		defer podmanTest.Podman([]string{"rm", "--time=0", "-f", "test"})
+		ctr := podmanTest.Podman([]string{"run", "-d", "--network", netName1, "--name", "test", ALPINE, "top"})
+		ctr.WaitWithDefaultTimeout()
+		Expect(ctr).Should(ExitCleanly())
+
+		exec1 := podmanTest.Podman([]string{"exec", "test", "ip", "addr", "show", "eth0"})
+		exec1.WaitWithDefaultTimeout()
+		Expect(exec1).Should(ExitCleanly())
+		Expect(exec1.OutputToString()).Should(ContainSubstring("eth0"))
+
+		defer podmanTest.removeNetwork(netName2)
+		nc2 := podmanTest.Podman([]string{"network", "create", netName2})
+		nc2.WaitWithDefaultTimeout()
+		Expect(nc2).Should(ExitCleanly())
+
+		// Inspect the network configuration
+		nic2 := podmanTest.Podman([]string{"network", "inspect", netName2, "--format", "{{.NetworkInterface}}"})
+		nic2.WaitWithDefaultTimeout()
+		Expect(nic2).Should(ExitCleanly())
+
+		// Once a container executes a new network, the nic will be
+		// created. We should clean those up best we can
+		defer deleteNetworkDevice(nic2.OutputToString())
+
+		conn := podmanTest.Podman([]string{"network", "connect", netName2, "test"})
+		conn.WaitWithDefaultTimeout()
+		Expect(conn).Should(ExitCleanly())
+		Expect(conn.ErrorToString()).Should(Equal(""))
+
+		exec2 := podmanTest.Podman([]string{"exec", "test", "ip", "addr", "show", "eth1"})
+		exec2.WaitWithDefaultTimeout()
+		Expect(exec2).Should(ExitCleanly())
+		Expect(exec2.OutputToString()).Should(ContainSubstring("eth1"))
+
+		rm := podmanTest.Podman([]string{"rm", "--time=0", "-f", "test"})
+		rm.WaitWithDefaultTimeout()
+		Expect(rm).Should(ExitCleanly())
+		Expect(rm.ErrorToString()).To(Equal(""))
```

Comment:
not needed as well

Reply:
I had to bring this back after deduplicating the tests by adding a loop. In the absence of "rm", the next iteration would conflict with the previous run and fail.